lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -512,7 +512,10 @@ MStatus MayaUsdProxyShapeBase::compute(const MPlug& plug, MDataBlock& dataBlock)
         ProxyAccessor::compute(_usdAccessor, plug, dataBlock);
         return retStatus;
     } else if (plug == outStageDataAttr) {
-        return computeOutStageData(dataBlock);
+        in_compute++;
```

Comment:
As discussed, for better encapsulation, future proofing (even though currently there is just one call site), and exception safeness (even though I don't think any exceptions can be thrown in computeOutStageData()), we can move the increment / decrement inside computeOutStageData(), with a guard object to take care of the decrement.